### PR TITLE
getting invites by recipient's social users and emails together

### PR DESCRIPTION
**Pull request by ymatsuda**
@andrewconner @leogrim 
I think I got this right, but please review (esp. InviteCommander).

### CODE REVIEW

**Comment by andrewconner** [comment]
@leogrim Mind checking this?

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/InviteCommander.scala [hunk 1]
```diff
@@ -176,16 +176,12 @@ class InviteCommander @Inject() (
...
-      val otherEmailInvites = for {
-        emailAccount <- userEmailAccounts if emailAccount.verified
-        invite <- invitationRepo.getByRecipientEmailAddress(emailAccount.address)
-      } yield (invite, SocialNetworks.EMAIL)
+      val networkTypes = userSocialAccounts.map { su => su.id.get -> su.networkType }.toMap
+      val verifiedEmails = userEmailAccounts.filter(_.verified)
+      val otherInvites = invitationRepo.getByRecipientSocialUserIdsAndEmailAddresses(userSocialAccounts.map(_.id.get).toSet, verifiedEmails.map(_.address).toSet)
 
+      val otherSocialInvites = otherInvites.filter(_.recipientSocialUserId.isDefined).map(invite => (invite, networkTypes(invite.recipientSocialUserId.get)))
+      val otherEmailInvites = otherInvites.filter(_.recipientEmailAddress.isDefined).map(invite => (invite, SocialNetworks.EMAIL))
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/InviteCommander.scala:184]: Sorry for the late review, I just saw this. This is consistent with the preexisting code :+1: 

Now it looks like the repo contains invites where both `recipientSocialUserId` and `recipientEmailAddress` are defined,   so these two sets have been overlapping. I'll take another look at the whole commander tomorrow to see if this was a problem, in which case I'll switch to `partition` here.

File: kifi-backend/modules/shoebox/app/com/keepit/model/InvitationRepo.scala [hunk 2]
```diff
@@ -123,6 +124,10 @@ class InvitationRepoImpl @Inject() (
     (for { row <- rows if row.recipientEmailAddress === emailAddress } yield row).list
   }
 
+  def getByRecipientSocialUserIdsAndEmailAddresses(socialUserInfoId: Set[Id[SocialUserInfo]], emailAddress: Set[EmailAddress])(implicit session: RSession): Seq[Invitation] = {
+    (for { row <- rows if row.recipientSocialUserId.inSet(socialUserInfoId) || row.recipientEmailAddress.inSet(emailAddress) } yield row).list
+  }
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/model/InvitationRepo.scala:129]: Great!

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala [hunk 3]
```diff
@@ -298,16 +298,9 @@ class AdminUserController @Inject() (
...
-    val byEmail: Seq[Id[User]] = emails map { email =>
-      invitationRepo.getByRecipientEmailAddress(email.address).map(_.senderUserId).flatten
-    } flatten
-    val all: Seq[Id[User]] = (byEmail ++ bySocial).toSet.toSeq
-    all map { userId =>
-      userRepo.get(userId)
-    }
+    val invites = invitationRepo.getByRecipientSocialUserIdsAndEmailAddresses(socialUserIds.toSet, emails.map(_.address).toSet)
+    val inviters = invites.map(_.senderUserId).flatten
+    userRepo.getAllUsers(inviters).values.toSeq
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala:303]: Cool, that's much simpler / cleaner / more efficient, thanks!

**Pull request merged**